### ISSUE TITLE
Feature request: update donation records in Civi when refunded in Stripe

### ISSUE THREAD

**Issue by laryn** [drastik/com.drastikbydesign.stripe#135]
In recent testing I had to do a bunch of live transactions, then refund them in Stripe, then update the status in CiviCRM. That got me thinking -- there's no reason not to use the 'charge.refunded' event from Stripe to update the status of the existing contribution in CiviCRM to "Refunded", is there? (Cancelling a recurring contribution/subscription already does something similar in pushing a change from Stripe back to CiviCRM).

(Also it made me wonder about the possibility of going the other direction and refunding a Stripe contribution just by changing status in CiviCRM, but that might be a whole other ballgame in terms of potential for problems.)

**Comment by h-c-c** [comment]
I agree, that get's old fast.  From an accounting standpoint, I'm wondering what happens to Stripe charges associated with full refunds in Civi.   It would be awesome if they weren't counted at all, since Stripe doesn't take fees in that event.   I'd have to look at what they do for partial refunds before starting this.

**Comment by richardsplayground** [comment]
+1 for this.  My use case is almost 100% around event registrations.  I end up having to do this double entry all the time when we cancel an event.

It seems like many people have different ways of triggering that a refund should be made in Civi.  There isn't just a clean "refund this contribution" button.  What I have done is in my price sets, I have an admin only-visible price set line item called Refund with a value of -1.  When I need to trigger a refund, I edit the contribution and I can then put in a quantity in that hidden refund field that would equal to a negative amount of what they already paid in the original contribution.  e.g. if their original contribution was $50, I put in 50 for the refund price field.  

Only then does Civi give me the option to "Process Refund" for that contribution.  When you click that button, it asks you what payment processor you used and a few other optional fields.

Just today I was doing this for about 25 transactions.  It's a lot of button clicking.  It would be amazing if Civi would automatically notice that you've refunded the transaction in Stripe.

None of this is hugely terrible for a site of our size but I can imagine a site with thousands or hundreds of thousands of users may feel more strongly about it.

Anyway, all of this to say, +1!  :)

**Comment by drastik** [comment]
I believe there are mechanisms for both, making either possible.
Update in Stripe -> Civi updated.
Update in Civi -> Stripe updated.

Regarding @richardsplayground 's comment, would like to know if there really is a "proper" way to refund in Civi.
